sugarscape.py

```diff
@@ -960,30 +965,61 @@ def verifyConfiguration(configuration):
+            if "all" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+                print(f"Cannot have selfishness factor range of {configuration['agentDecisionModelTribalFactor']}. Disabling agent selfishness.")
+        configuration["agentDecisionModelTribalFactor"] = [-1, -1]
+    elif configuration["agentDecisionModelTribalFactor"][1] > 1:
+        if "all" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+            print(f"Cannot have selfishness factor range of {configuration['agentDecisionModelTribalFactor']}. Capping maximum at 1.0.")
+        configuration["agentDecisionModelTribalFactor"][1] = 1
 
     if configuration["agentTagStringLength"] < 0:
+        if "all" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+            print(f"Cannot have a negative tag string length. Setting tag string length to 0.")
         configuration["agentTagStringLength"] = 0
     if configuration["environmentMaxTribes"] < 0:
+        if "all" in configuration["debugMode"] or "environment" in configuration["debugMode"]:
+            print(f"Cannot have a negative number of tribes. Setting tag string length to 0.")
         configuration["environmentMaxTribes"] = 0
+    
     # Ensure at most number of tribes is equal to agent tag string length
     if configuration["agentTagStringLength"] > 0 and configuration["environmentMaxTribes"] > configuration["agentTagStringLength"]:
+        if "all" in configuration["debugMode"] or "environment" in configuration["debugMode"] or "agent" in configuration["debugMode"]:
+            print(f"Cannot have a longer tag string length than maximum number of tribes. Setting \"environmentMaxTribes\" equal to \"agentTagStringLength\".")
         configuration["environmentMaxTribes"] = configuration["agentTagStringLength"]
 
-    configuration["agentDecisionModelTribalFactor"].sort()
-    if configuration["agentDecisionModelTribalFactor"][1] > 1:
-        configuration["agentDecisionModelTribalFactor"][1] = 1
-    elif configuration["agentDecisionModelTribalFactor"][0] < 0:
-        configuration["agentDecisionModelTribalFactor"][0] = -1
-
-    # Ensure at most number of tribes is equal to the number of colors in the GUI
-    maxTribes = 25
-    if configuration["environmentMaxTribes"] > maxTribes:
+    # Ensure at most number of tribes and decision models are equal to the number of colors in the GUI
+    maxColors = 25
+    if configuration["environmentMaxTribes"] > maxColors:
         if "all" in configuration["debugMode"] or "sugarscape" in configuration["debugMode"] or "environment" in configuration["debugMode"]:
-            print(f"Cannot provide {configuration['environmentMaxTribes']} tribes. Allocating maximum of {maxTribes}.")
-        configuration["environmentMaxTribes"] = maxTribes
+            print(f"Cannot provide {configuration['environmentMaxTribes']} tribes. Allocating maximum of {maxColors}.")
```

Comment:
Definitely a nitpick, but we've been using double quotes to access keys. @nkremerh can advise if single quotes are OK or not.

Reply:
Using single quotes in a fstring won't properly parse.

Reply:
That's funny, it looks like my IDE is more lenient with fstring parsing than other ones. You're probably right—better to keep them as single quotes.